applications/adv_networking_bench/cpp/testing/test_ano_bench.py

```diff
@@ -0,0 +1,165 @@
+    """
+    # Get the first two NICs for this test
+    tx_interface, rx_interface = nvidia_nics[0], nvidia_nics[1]
+
+    # Prepare config
+    config_file = os.path.join(work_dir, "adv_networking_bench_default_tx_rx.yaml")
+    try:
+        update_yaml_file(
+            config_file,
+            config_file,
+            {
+                "scheduler.max_duration_ms": 10000,
+                "advanced_network.cfg.interfaces[0].address": tx_interface.bus_id,
+                "advanced_network.cfg.interfaces[1].address": rx_interface.bus_id,
+                "bench_tx.eth_dst_addr": rx_interface.mac_address,
+                "bench_tx.address": tx_interface.bus_id,
+            },
+        )
+    except (ValueError, IOError) as e:
+        pytest.fail(f"Failed to update YAML configuration: {e}")
+
+    # Build the command
+    executable = os.path.join(work_dir, "adv_networking_bench")
+    command = f"{executable} {config_file}"
+
+    # Run the application until completion and parse the results
+    result = run_command(command, stream_output=True)
+    results = parse_benchmark_results(result.stdout + result.stderr)
+
+    # Validate some expected metrics
+    port_map = {0: 1}  # Port 0 (TX) sends to Port 1 (RX)
+    avg_throughput_threshold = 90.0
+    missed_pkts_threshold = 0.1
+    error_pkts_threshold = 0.0
+    assert results.validate_missed_packets(
+        port_map, missed_pkts_threshold
+    ), "Missed packets validation failed"
+    assert results.validate_errored_packets(
+        port_map, error_pkts_threshold
+    ), "Errored packets validation failed"
+    assert results.validate_throughput(
+        port_map, avg_throughput_threshold
+    ), "Throughput validation failed"
+
+
+def test_multi_rx_q(work_dir, nvidia_nics):
+    """
+    Test 2: RX multi queue with a single CPU core using scapy to send packets.
+
+    This test focuses on:
+    - Validating the distribution of packets to RX queues
+    """
+    # Get the NICs for this test
+    tx_interface, rx_interface = nvidia_nics[0], nvidia_nics[1]
+
+    # Prepare config
+    config_file = os.path.join(work_dir, "adv_networking_bench_default_rx_multi_q.yaml")
+    try:
+        update_yaml_file(
+            config_file,
+            config_file,
+            {
+                "scheduler.max_duration_ms": 10000,
+                "advanced_network.cfg.interfaces[0].address": rx_interface.bus_id,
+            },
+        )
+    except (ValueError, IOError) as e:
+        pytest.fail(f"Failed to update YAML configuration: {e}")
+
+    # Test parameters
+    expected_q_pkts = {0: 1, 1: 1}  # Expecting 1 packet for queue 0 and 1
+
+    # Build the absolute path to the executable
+    executable = os.path.join(work_dir, "adv_networking_bench")
+    command = f"{executable} {config_file}"
+
+    # Run the application (non-blocking)
+    p = start_process(command)
+
+    # Send packets with scapy after a 5s delay
+    def send_test_packets():
```

Comment:
I liked this in a scapy script because it's possibly this section may grow a lot in the future, and the scapy files constrain the code from blowing up.

Reply:
We could still do it, would need to pass arguments to it though (like the NIC name/address which was creating issues here). I could see another module called `scapy_utils.py` with multiple jobs that are convenient to use and reuse.

My suggestion would be to wait until we see a clear need if this doesn't scale up, because currently it's isolated to this test.